tests/components/demo/test_vacuum.py

```diff
@@ -47,326 +47,293 @@
-        assert (
-            "Executing test_command({'p1': 3})"
-            == new_state_complete.attributes[ATTR_STATUS]
-        )
+@pytest.fixture(autouse=True)
+async def setup_demo_vacuum(hass):
+    """Initialize setup demo vacuum."""
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_PLATFORM: "demo"}})
+
+
+async def test_supported_features(hass):
+    """Test vacuum supported features."""
+    state = hass.states.get(ENTITY_VACUUM_COMPLETE)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 2047
+    assert state.attributes.get(ATTR_STATUS) == "Charging"
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) == "medium"
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) == FAN_SPEEDS
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_MOST)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 219
+    assert state.attributes.get(ATTR_STATUS) == "Charging"
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) is None
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_BASIC)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 195
+    assert state.attributes.get(ATTR_STATUS) == "Charging"
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) is None
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_MINIMAL)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 3
+    assert state.attributes.get(ATTR_STATUS) is None
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) is None
+    assert state.attributes.get(ATTR_FAN_SPEED) is None
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 0
+    assert state.attributes.get(ATTR_STATUS) is None
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) is None
+    assert state.attributes.get(ATTR_FAN_SPEED) is None
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 13436
+    assert state.state == STATE_DOCKED
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) == "medium"
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) == FAN_SPEEDS
+
+
+async def test_methods(hass):
+    """Test if methods call the services as expected."""
+    hass.states.async_set(ENTITY_VACUUM_BASIC, STATE_ON)
+    await hass.async_block_till_done()
+    assert vacuum.is_on(hass, ENTITY_VACUUM_BASIC)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5315232)